test/binary-elementwise-operator-tester.cc

```diff
@@ -724,6 +724,159 @@ void BinaryElementwiseOperatorTester::TestF16() const {
+        0, std::numeric_limits<int16_t>::max());
+  // Compute generalized shapes.
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> input1_dims;
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> input2_dims;
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> output_dims;
+  std::fill(input1_dims.begin(), input1_dims.end(), 1);
+  std::fill(input2_dims.begin(), input2_dims.end(), 1);
+  std::fill(output_dims.begin(), output_dims.end(), 1);
+  std::copy(input1_shape().cbegin(), input1_shape().cend(),
+            input1_dims.end() - num_input1_dims());
+  std::copy(input2_shape().cbegin(), input2_shape().cend(),
+            input2_dims.end() - num_input2_dims());
+  for (size_t i = 0; i < XNN_MAX_TENSOR_DIMS; i++) {
+    if (input1_dims[i] != 1 && input2_dims[i] != 1) {
+      ASSERT_EQ(input1_dims[i], input2_dims[i]);
+    }
+    output_dims[i] = std::max(input1_dims[i], input2_dims[i]);
+  }
+  const size_t num_output_elements =
+      std::accumulate(output_dims.begin(), output_dims.end(), size_t(1),
+                      std::multiplies<size_t>());
+
+  // Compute generalized strides.
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> input1_strides;
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> input2_strides;
+  std::array<size_t, XNN_MAX_TENSOR_DIMS> output_strides;
+  size_t input1_stride = 1, input2_stride = 1, output_stride = 1;
+  for (size_t i = XNN_MAX_TENSOR_DIMS; i != 0; i--) {
+    input1_strides[i - 1] = input1_dims[i - 1] == 1 ? 0 : input1_stride;
+    input2_strides[i - 1] = input2_dims[i - 1] == 1 ? 0 : input2_stride;
+    output_strides[i - 1] = output_stride;
+    input1_stride *= input1_dims[i - 1];
+    input2_stride *= input2_dims[i - 1];
+    output_stride *= output_dims[i - 1];
+  }
+
+  std::vector<int16_t> input1(XNN_EXTRA_BYTES / sizeof(int16_t) +
+                            num_input1_elements());
+  std::vector<int16_t> input2(XNN_EXTRA_BYTES / sizeof(int16_t) +
+                            num_input2_elements());
+  std::vector<int16_t> output(num_output_elements);
+  std::vector<int16_t> output_ref(num_output_elements);
+  float output_float;
+  int32_t res;
+  for (size_t iteration = 0; iteration < iterations(); iteration++) {
+    std::generate(input1.begin(), input1.end(), [&]() {return s16dist(rng);});
+    std::generate(input2.begin(), input2.end(), [&]() {return s16dist(rng);});
+    std::fill(output.begin(), output.end(), 0xA5);
+    float scale = (input1_scale() * input2_scale())/ output_scale();
+    // Compute reference results.
+    for (size_t i = 0; i < output_dims[0]; i++) {
+      for (size_t j = 0; j < output_dims[1]; j++) {
+        for (size_t k = 0; k < output_dims[2]; k++) {
+          for (size_t l = 0; l < output_dims[3]; l++) {
+            for (size_t m = 0; m < output_dims[4]; m++) {
+              for (size_t n = 0; n < output_dims[5]; n++) {
+                output_float = static_cast<float>(Compute(
+                        static_cast<int32_t>(input1[i * input1_strides[0] + j * input1_strides[1] +
```

Comment:
I think the implementation of this operator is prone to overflow, and this test is prone to overflow in the same way (so the test passes incorrectly). I think this should cast the inputs to float rather than int32_t, so we call `Compute(float, float)` instead of `Compute(int32_t, int32_t)`, which I think will uncover this overflow issue.

Reply:
changed to compute(float,float) matching with qs8 test